sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -213,6 +215,56 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
 
   }
 
+  test("persist and unpersist") {
```

Comment:
I'd consider putting these in their own suite.  Caching is a pretty isolated concern.  I would also like to see some more tests for operations that aren't expressed as logical operations (i.e. map/filter with lambda functions instead of expressions).